### PR TITLE
Requesting Signoff for Identification System

### PR DESCRIPTION
**Pull request by DillonSW**

### CODE REVIEW

File: Documentation/Signoffs/Barcode_Scanner.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Barcode Scanner Subsystem  
+
+## Subsystem Function  
+
+This Subsystem's role in the team's design is to scan barcodes on school devices before being placed into the machine. This is done to number a device in each slot so a student can be assigned the number when renting the device, and is also  done to assure that devices are accounted for.  
+
+## Constraints
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Barcode_Scanner.md]: Incomplete constraints. can't design without clearly identifying the target.

**Comment by DillonSW** [comment]
Requesting Signoff for Identification Subsystem #2

File: Documentation/Signoffs/IdentificationSystem.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+## Constraints  
+
+* Barcode Scanners shall be low power and consume no greater than 5 V 0.5 A. This is due to being powered by the miniPC. Common computers only output a max rating of 5 V 0.5 A.  
+^ Source: https://resources.pcb.cadence.com/blog/2020-what-are-the-maximum-power-output-and-data-transfer-rates-for-the-usb-standards  
+
+* Barcode Scanners shall be able to decode Code-128 barcodes 
+
+* RFID Card Scanner shall be low power and consume no greater than 5 V 0.5 A   
+
+* RFID Card Scanner shall operate on 13.56 MHz commonly used for higher security access badges
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/IdentificationSystem.md]: Where does this come from? Is this the spec for the eagle card?

File: Documentation/Signoffs/IdentificationSystem.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+* Barcode Scanners shall be low power and consume no greater than 5 V 0.5 A. This is due to being powered by the miniPC. Common computers only output a max rating of 5 V 0.5 A.  
+^ Source: https://resources.pcb.cadence.com/blog/2020-what-are-the-maximum-power-output-and-data-transfer-rates-for-the-usb-standards  
+
+* Barcode Scanners shall be able to decode Code-128 barcodes 
+
+* RFID Card Scanner shall be low power and consume no greater than 5 V 0.5 A   
+
+* RFID Card Scanner shall operate on 13.56 MHz commonly used for higher security access badges  
+
+* Barcode Scanners and the RFID Card reader shall be powered by a USB port to communicate with the miniPC
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/IdentificationSystem.md]: redundant and repetitive?

File: Documentation/Signoffs/IdentificationSystem.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
+# Identification Subsystem  
+
+## Subsystem Function  
+
+This Subsystem's role in the team's design is to scan barcodes on school devices before being placed and when being taken from the machine. The subsystem also includes a RFID Card Reader to verify and store students' identity. The PC will then receive information from the modules and store information in a database for our customer. 
+
+## Constraints
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/IdentificationSystem.md:7]: Excellent structure and layout. These get right to the point. This is good.

File: Documentation/Signoffs/IdentificationSystem.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+* Barcode Scanners shall be able to decode Code-128 barcodes 
+
+* RFID Card Scanner shall be low power and consume no greater than 5 V 0.5 A   
+
+* RFID Card Scanner shall operate on 13.56 MHz commonly used for higher security access badges  
+
+* Barcode Scanners and the RFID Card reader shall be powered by a USB port to communicate with the miniPC  
+
+* A Barcode Scanner and Card Reader will be monitered by USB voltmeter to ensure correct operation
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/IdentificationSystem.md:22]: Not a constraint really. Just a design detail.

File: Documentation/Signoffs/IdentificationSystem.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+* A Barcode Scanner and Card Reader will be monitered by USB voltmeter to ensure correct operation  
+
+* Barcodes shall be of decent quality to follow ISO 15416 standard. Assures that the codes can be read by any scanner  
+
+* The MiniPC shall run on Windows to provide the needed applications for PLC and Database
+
+* The MiniPC shall occupy no larger than a 6x6 in. space to fit in the machine
+
+* The MiniPC will provide an ethernet port to communicate with the PLC
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/IdentificationSystem.md]: Probably need more than one NIC (network interface card). You can get external, usb based NICs if the miniPC doesn't have 2.

File: Documentation/Signoffs/IdentificationSystem.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+* Shall include a USB Hub to provide enough ports for modules
+
+* Shall include a port to connect to the UI system  
+
+## Schematic
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/IdentificationSystem.md]: What is the miniPC? is this a specific device? (Probably not a problem and simply a misunderstanding on my part)

Comment (JesseTNRoberts) [reply Documentation/Signoffs/IdentificationSystem.md]: How will this communicate with the PLC? What does it need to communicate? Communication is going to be the hardest part of this system.

**Review by JesseTNRoberts — COMMENTED** [review]
The communication between the PC and PLC will be less than straightforward. It won't be as simple as an ethernet cable. What protocol is to be used?

**Review by JesseTNRoberts — COMMENTED** [review]
Still does not address communication from PLC to PC.

**Comment by DillonSW** [comment]
I am unsure of what is not addressed. Do you want a step by step explanation of connecting the PLC to the PC using the TIA Portal Software?

File: Documentation/Signoffs/IdentificationSystem.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+^ Source: https://us.profinet.com/profinet-explained/  
+
+To connect the PLC to the PC, the TIA Portal software must first be purchased and downloaded. After creating a new project, the device we are using will be added from the available library. The exact CPU is listed in the PLC we received. There is also an option to detect our PLC if we cannot find the device. The portal will use a PN/IE interface and the USB NIC as the adapter for connection. To write values from the sensors, an offset is needed. The common offset of the address is 10. Our team is giving full access to the PLC for the project. Once the PLC is set up, the CPU can be started to run the program. When our program is created, there will need to be a way to give the PLC a command from our code to TIA portal. With the newer versions of TIA portal, python can send data from the program to the portal for the PLC to execute. The data will go through Modbus to reach the TIA portal. Python has libraries such as Pymodbus to connect. This will be used to control the motor and LED indicators. Data can also be read from the PLC if needed.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/IdentificationSystem.md]: This won't allow you to programmatically interact with the PLC from the PC and vise versa. What you have described is the software needs of the computer used to program the PLC. That is not the same.

File: Documentation/Signoffs/IdentificationSystem.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+## Schematic  
+
+![Schematic](https://github.com/DillonSW/Capstone_Team_5/blob/Team5-signoff-Barcode-Scanner/images/ID_System.jpg?raw=true)    
+
+The Schematic above shows the layout of the ID subsystem. The PC connects to the PLC through a USB NIC, which is a USB to ethernet adapter, and is powered by the power subsystem. The PC is connected to and allows for multiple modules through six USB 3.0 ports. Two USB Voltmeters are used for a Barcode Scanner and RFID Reader. The UI system is connected to the MiniPC through the HDMI on the PC.  
+
+## Analysis  
+
+Commonly a barcode scanner is made up of three parts: illumination system, sensor system, and decoder. The illumination system is responsible for emiting a red light onto the code which seperates the dark and light sections. The sensor detects the reflected light from the white portion of the code and generates an analog signal which is then sent to the decoder. The decoder will translate the signal into text and checks the information with the check digit. The text is then finally sent to the computer connected to the device to be read.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/IdentificationSystem.md]: Why is this talked about? how is this relevant?

**Review by JesseTNRoberts — COMMENTED** [review]
The constraints make sense and the schematic looks good. The analysis needs work. It includes much extraneous and unnecessary information. It should clearly show how you know that the design will meet your expectations (the constraints).

File: Documentation/Signoffs/IdentificationSystem.md [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+
+* The PC will provide at minimum an Intel Core i5-8400H or stronger CPU and 16GB RAM to support PLC programs
+
+## Schematic  
+
+![Schematic](https://github.com/DillonSW/Capstone_Team_5/blob/Team5-signoff-Barcode-Scanner/images/ID_System.jpg?raw=true)    
+
+The Schematic above shows the layout of the ID subsystem. The PC connects to the PLC through a USB NIC, which is a USB to ethernet adapter, and is powered by the power subsystem. The PC is connected to and allows for multiple modules through six USB 3.0 ports. Two USB Voltmeters are used for a Barcode Scanner and RFID Reader. The UI system is connected to the MiniPC through the HDMI on the PC.  
+
+## Analysis
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/IdentificationSystem.md:36]: Address these in the same order as the constraints and state what it is your are analyzing.

**Review by JesseTNRoberts — APPROVED** [review]
Verify that the scanners will work with ubuntu before purchasing.
